### ISSUE TITLE
Client Compilation

### ISSUE THREAD

**Issue by jescalan** [pugjs/pug#2407]
Just wanted to see if anyone had a moment to quickly describe the status of `compileClient` in pug. I pulled down the alpha and messed with it a little bit, only to find that things were quite different than they are in current jade.

What I noticed so far:

- No more `runtime.js`, but there is a `pug-runtime` library. Not entirely sure if it's still needed to run a client-compiled file. It seems like it is not, from my testing so far.
- Client-compiled strings are exported along with some helpers, as a named function called `template`, which can be changed using a `templateName` option. However, this option doesn't actually work and it's always exported as `template`.
- If you prefix a client template with a `module.exports =` it no longer works with a commonjs-based system, since it exports one of the helpers instead.
- If every client compile is exported along with its helpers, this is going to be very inefficient if you are trying to put multiple templates on a page. I thought this was the reason for the runtime library?
- Instead of using characters like `<` it uses unicode escape sequences like `\u003C`. This looks very strange when written to a file. It also does not render correctly when written to a page, maybe this is a bug?

It's possible I'm doing something terribly wrong, but none of the docs or upgrade discussions that I have come across have discussed it, so any push in the right direction would be helpful!

**Comment by TimothyGu** [comment]
Those are just about the changes done. I'd love it if you could help prepare some documentation for using client mode @[pug-en](https://github.com/pugjs/pug-en). Just add a src/api/client.md and follow other src/api/*.md's syntax.

**Comment by strand** [comment]
@TimothyGu It looks like the client API documentation still needs to be written in @pug-en. If I submit a draft PR, will you review and advise me on meeting pug's standards.

**Comment by ForbesLindesay** [comment]
Yes, either @TimothyGu or me will review any pull request to the pug-en documentation.

**Comment by matthewjumpsoffbuildings** [comment]
What is the status of this? The docs still seem out of date, Im not sure how to implement client side compilation currently...

Im using webpack, no idea how to get my JS files to reference the pug and correctly precompile the templates to functions etc
